### ISSUE TITLE
Tooltips flicker

### ISSUE THREAD

**Issue by Brandhand** [SawfishWM/sawfish#45]
I have a problem in firefox, since 81.0 I guess, concerning flickering
tooltips in the render-area. It's about the 'internal' tooltips
rendered because of title attributes in html tags. The tooltips in the
GUI area are pefectly fine.

Try for instance https://heise.de and hover over the buttons 
  Topthemen: iPhone 12  E-Auto  Security

The tooltips are just visible for less than a tenth of a second and
they are almost under the cursor not below.

I first blamed firefox. But I tried other WMs (metacity, openwm) and
they behave correctly.

I used sawfish 1.11.90 and now upgraded to 1.12.90 from phtonic
repository without any changes.

I also found out, that Firefox is somewhat to blame, too, because the
current ESR-78.4 behaves allright under sawfish.

I hope someone can help.

**Comment by RobertZenz** [comment]
No repro here with Ubuntu 18.04, Firefox 81.0.2 and Sawfish 1.11.90. The tooltips show up and stay. I'm using compton as compositor.

**Comment by Brandhand** [comment]
Thanks for the respond. I tried Firefox 81, pulled the packages from
Ubuntu 18.04, installed compton, nothing helped.

Then I stumbled over sawfish-themes, and look, the theme Crux is the
one to blame.

I found out that at least on my system the themes

Anonymous
Dust 
Crux
gradient-tabbed
Shiki
StyleTab
Zami-like

have the issue I described above, and

absolute-e
candido
Elberg-tabbed
microGUI
mxflat
typo

are running just fine. Or at least almost fine because the tooltip is
still under the cursor covering a bit the tooltip. But I'm fine with
it.

If anybody could confirm this - you have to restart firefox or sawfish
on any theme-change - somebody may fix it or at least pull the themes
from the repository.

**Comment by RobertZenz** [comment]
You are right, I'm using mxflat butI can reproduce it with mxflat. I can also reproduce it wih mxflat when I enable "Draw an extra window border" with a width of 6 pixels, so this isn't a graphical issue as I presumed, it's window placement issue. The displayed tooltip covers the area on which the mouse hovers, which triggers a mouse leave (or so) in Firefox which assumes that the mouse was moved away from the element or even browser window, which removes the tooltip.

However, the extra border is not actually added to the tooltip, but the position of the tooltip is shifted by the amount instead. Seems like calculating the tooltip window position is off by the set border, even though it is not displayed or used. That might actually be a bug in Sawfish.
